subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/TestRunnerImpl.kt

```
@@ -66,9 +67,27 @@ internal class TestRunnerImpl(
 
                     val expectedResultsCount = tests.count()
 
+                    val gottenResultsByDeviceName : MutableMap<String, MutableList<TestRunResult>> = mutableMapOf()
+                    val expectedResultsCountByDeviceName = tests
+                        .groupBy { it.deviceName.name }
+                        .map { (deviceName, tests) ->
+                            gottenResultsByDeviceName[deviceName] = mutableListOf()
+                            deviceName to tests.count()
+                        }
+
                     val gottenResults = mutableListOf<TestRunResult>()
                     for (result in state.results) {
                         gottenResults.add(result)
+                        gottenResultsByDeviceName[result.request.testCase.deviceName.name]?.add(result)
```

We should fail fast when we can't find results by deviceName